src/DeclarationProvider.ts

```diff
@@ -190,14 +194,41 @@ export class DeclarationProvider implements Disposable {
         let funcEndChar: number;
         let mDefs = {};
         console.log('READ DECLARATIONS');
+        let oldNamespaces = this.fileNamespaces.get(uri);
+        if (oldNamespaces) {
+            for (let key of oldNamespaces.keys()) {
+                let ns = this.allNamespaces.get(key);
+                if (ns && ns.uri === uri) {
+                    this.allNamespaces.delete(key);
+                }
+            }
+        }
+        this.fileNamespaces.delete(uri);
+
+        let oldClasses = this.fileClasses.get(uri);
+        if (oldClasses) {
+
+            for (let key of oldClasses.keys()) {
+                let clazz = this.allClasses.get(key);
+                if (clazz && clazz.uri === uri) {
+                    this.allClasses.delete(key);
+                }
+            }
+        }
+        this.fileClasses.delete(uri);
+
+        let namespaces = new Set<string>();
+        let classes = new Set<string>();
+        let namespaceSymbol: BrightScriptDeclaration | null;
+        let classSymbol: BrightScriptDeclaration | null;
 
         for (const [line, text] of iterlines(input)) {
             // console.log("" + line + ": " + text);
             funcEndLine = line;
             funcEndChar = text.length;
 
             //FUNCTION START
-            let match = /^\s*(?:function|sub)\s+(.*[^\(])\s*\((.*)\)/i.exec(text);
+            let match = /^\s*(?:override)*\s*(?:public|private)*\s*(?:function|sub)\s+(.*[^\(])\s*\((.*)\)/i.exec(text);
```

Comment:
@georgejecook 

The syntax should be (pseudocode):
optional AccessModifier (`public`|`protected`|`private`)
optional `override`
`sub`|`function`
FunctionName

Examples:
```
public override sub GetName()
protected override sub GetName()
private override sub GetName()
public sub GetName()
protected sub GetName()
private sub GetName()
sub GetName()
```